packages/eslint-plugin-office-addins/src/rules/no-empty-load.ts

```diff
@@ -0,0 +1,79 @@
+import { TSESTree } from "@typescript-eslint/experimental-utils";
+import {
+  Reference,
+  Scope,
+  Variable,
+} from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+import { isGetFunction, isLoadFunction } from "../utils";
+
+export = {
+  name: "no-empty-load",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"problem",
+    messages: {
+      emptyLoad: "Calling load without any argument can slow down your add-in",
+    },
+    docs: {
+      description:
+        "Calling load without any argument can cause load unneeded data and slow down your add-in",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Best Practices",
+      recommended: <false | "error" | "warn">false,
+      url: "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model?view=powerpoint-js-1.1#calling-load-without-parameters-not-recommended",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
+    function isEmptyLoad(node: TSESTree.MemberExpression): boolean {
+      return (
+        isLoadFunction(node) &&
+        node.parent?.type == TSESTree.AST_NODE_TYPES.CallExpression &&
+        node.parent.arguments.length === 0
+      );
+    }
+
+    function findEmptyLoad(scope: Scope) {
+      scope.variables.forEach((variable: Variable) => {
+        let getFound: boolean = false;
+        variable.references.forEach((reference: Reference) => {
+          const node: TSESTree.Node = reference.identifier;
+
+          if (
+            reference.isWrite() &&
+            reference.writeExpr &&
+            isGetFunction(reference.writeExpr)
+          ) {
+            getFound = false; // In case of reassignment
+            if (isGetFunction(reference.writeExpr)) {
```

Comment:
You'll need "reference.writeExpr" as part of this conditional.

Reply:
Added a test case for this condition and I think now it is fixed

Reply:
As is it works . . . the only question is weather you need to check reference.writeExpr in the outer if or the inner if.  As far as I can tell it's needed for the isGetFunction call to work, but do you have another reason for checking it earlier?

Reply:
I guess inner makes more sense as well, just updated it